repos/spack_repo/builtin/packages/dbcsr/package.py

```diff
@@ -197,6 +204,7 @@ def cmake_args(self):
                 "gfx90a": "Mi250",
                 "gfx90a:xnack-": "Mi250",
                 "gfx90a:xnack+": "Mi250",
+                "gfx942": "Mi250",
```

Comment:
```suggestion
                "gfx942": "Mi300",
```
?

Reply:
The dbcsr build system does not support Mi300 yet for various reasons. the `"gfx942": "Mi250",` is needed here :(

Reply:
Ah, I see. In this case I think it would be better to have a conflict on `amdgpu_target=gfx942` then.

Reply:
> The lack of optimized parameters for Mi300 should not block users from using dbcsr.

Users will not be blocked to use `dbcsr`, but they will have to set `dbcsr amdgpu_target=gfx90a`. In this way, you will not magically change the meaning of setting `amdgpu_target=gfx942` once the architecture is properly supported. In this way the user is consciously using the parameters for another architecture (you can also add this suggestion to the conflict message).

Something like
```
cp2k +rocm amdgpu_target=gfx942 ^dbcsr +rocm amdgpu_target=gfx942
```
should work, without the need to change the recipe.

Reply:
added a conflict and a message suggesting to use `gfx90a` when dbcsr is compiled for mi300. 
NB: Using parameters from older architecture should not lead to a bad behavior otherwise it means the code is buggy.